src/sync.js

```diff
@@ -508,10 +511,30 @@ SyncApi.prototype._sync = function(syncOptions) {
     // normal timeout= plus buffer time
     const clientSideTimeoutMs = this.opts.pollTimeout + BUFFER_PERIOD_MS;
 
-    debuglog('Starting sync since=' + syncToken);
-    this._currentSyncRequest = client._http.authedRequest(
-        undefined, "GET", "/sync", qps, undefined, clientSideTimeoutMs,
-    );
+    let isCachedResponse = false;
+    if (self.opts.syncAccumulator && !syncOptions.hasSyncedBefore) {
+        let data = self.opts.syncAccumulator.getJSON();
+        // Don't do an HTTP hit to /sync. Instead, load up the persisted /sync data,
+        // if there is data there.
+        if (data.nextBatch) {
+            debuglog("sync(): not doing HTTP hit, instead returning stored /sync data");
+            // We must deep copy the stored data so that the /sync processing code doesn't
+            // corrupt the internal state of the sync accumulator (it adds non-clonable keys)
+            data = JSON.parse(JSON.stringify(data));
```

Comment:
I know it's the same thing, but https://github.com/matrix-org/matrix-js-sdk/blob/master/src/utils.js#L254

(ie. if it's there, we should either be consistent about using it or get rid if it)